Sources/Datadog/RUM/Instrumentation/Actions/SwiftUI/SwiftUIActionModifier.swift

```diff
@@ -0,0 +1,103 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+#if canImport(SwiftUI)
+import SwiftUI
+
+/// `SwiftUI.ViewModifier` for RUM which invoke `addUserAction` from the
+/// global RUM Monitor when the modified view receives a tap.
+@available(iOS 13, *)
+internal struct RUMTapActionModifier: SwiftUI.ViewModifier {
+    /// The required number of taps to complete the tap action.
+    let count: Int
+
+    /// Action Name used for RUM Explorer.
+    let name: String
+
+    /// Custom attributes to attach to the Action.
+    let attributes: [AttributeKey: AttributeValue]
+
+    func body(content: Content) -> some View {
+        content.simultaneousGesture(
+            TapGesture(count: count).onEnded { _ in
+                Global.rum.addUserAction(type: .tap, name: name, attributes: attributes)
+            }
+        )
+    }
+}
+
+/// `SwiftUI.ViewModifier` for RUM which invoke `addUserAction` from the
+/// global RUM Monitor when the modified view receives a swipe.
+@available(iOS 13, *)
+internal struct RUMSwipeActionModifier: SwiftUI.ViewModifier {
+    /// The minimum dragging distance before the gesture succeeds.
+    let minimumDistance: CGFloat
+
+    /// Action Name used for RUM Explorer.
+    let name: String
+
+    /// Custom attributes to attach to the Action.
+    let attributes: [AttributeKey: AttributeValue]
+
+    /// Start state of the drag gesture.
+    @State private var started = false
+
+    func body(content: Content) -> some View {
+        content.simultaneousGesture(
+            DragGesture(minimumDistance: minimumDistance, coordinateSpace: .local)
+                .onChanged { _ in
+                    if !started {
+                        started.toggle()
+                        Global.rum.stopUserAction(type: .swipe, name: name, attributes: attributes)
+                    }
+                }
+                .onEnded { _ in
+                    if started {
+                        Global.rum.stopUserAction(type: .swipe, name: name, attributes: attributes)
+                        started.toggle()
+                    }
+                }
+        )
+    }
```

Comment:
This is a proposal for instrumentation `swipe` and/or `scroll` actions. SwiftUI only provide a `DragGesture`.

Reply:
This is not covered by tests. If we adopt this, we should add tests before merging